### PR TITLE
ISSUES-5 Support INSERT INTO With TCP Protocol

### PR DESCRIPTION
**Pull request by zhang2014**
#5

### CODE REVIEW

**Comment by zhang2014** [comment]
almost done

File: src/main/java/org/houseflys/jdbc/data/DataTypeFactory.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        creators.put("Int8", new DataTypeInt8());
+        creators.put("Int16", new DataTypeInt16());
+        creators.put("Int32", new DataTypeInt32());
+        creators.put("Int64", new DataTypeInt64());
+        creators.put("UInt8", new DataTypeInt8());
+        creators.put("UInt16", new DataTypeInt16());
+        creators.put("UInt32", new DataTypeInt32());
+        creators.put("UInt64", new DataTypeInt64());
+        creators.put("String", new DataTypeString());
+        creators.put("Enum8", new DataTypeString());
```
Comment (sundy-li) [inline src/main/java/org/houseflys/jdbc/data/DataTypeFactory.java]: should delete these two lines,  Enum8 and Enum16 are complex data type

File: src/main/java/org/houseflys/jdbc/data/type/complex/DataTypeDateTime.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Override
+    public Object deserializeTextQuoted(QuotedLexer lexer) throws SQLException {
+        QuotedToken token = lexer.next();
+        Validate.isTrue(token.type() == QuotedTokenType.StringLiteral, "");
+        String timestampString = token.data();
+
+        String[] dateAndTime = timestampString.split(" ", 2);
+        String[] yearMonthDay = dateAndTime[0].split("-", 3);
+        String[] houseMinuteSecond = dateAndTime[1].split(":", 3);
```
Comment (sundy-li) [inline src/main/java/org/houseflys/jdbc/data/type/complex/DataTypeDateTime.java]: spell check,  ` hours `

File: src/test/java/org/houseflys/jdbc/AbstractITest.java [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 public abstract class AbstractITest {
 
-    private static final int SERVER_PORT = Integer.parseInt(System.getProperty("CLICK_HOUSE_SERVER_PORT"));
+    private static final int SERVER_PORT = 9000;//Integer.parseInt(System.getProperty("CLICK_HOUSE_SERVER_PORT"));
```
Comment (sundy-li) [inline src/test/java/org/houseflys/jdbc/AbstractITest.java]: I think it is better to use 

```java

 private static int SERVER_PORT;
 static{
        SERVER_PORT = 9000；
        if(System.getProperty("ClickHousePort") != null){
            SERVER_PORT = Integer.parseInt(System.getProperty("CLICK_HOUSE_SERVER_PORT"));
        }
    }

```

File: src/main/java/org/houseflys/jdbc/ClickHouseStatement.java [hunk 4]
```diff
@@ -2,38 +2,51 @@
...
 import java.sql.SQLException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-import org.houseflys.jdbc.protocol.QueryResponse;
+import org.houseflys.jdbc.stream.ValuesInputFormat;
 import org.houseflys.jdbc.wrapper.SQLStatement;
 
 public class ClickHouseStatement extends SQLStatement {
+    private static final Pattern VALUES_REGEX = Pattern.compile("[V|v][A|a][L|l][U|u][E|e][S|s]\\(");
```
Comment (sundy-li) [inline src/main/java/org/houseflys/jdbc/ClickHouseStatement.java]: may have blank chars before '(',  should use \\s*\\(

What about: select * from test where name = 'VALUES (a,b,c,d)' ?
It's not an insert query, but matches the Pattern,   should use `QuotedLexer` to check it's an insert query.

Comment (zhang2014) [reply src/main/java/org/houseflys/jdbc/ClickHouseStatement.java]: > What about: select * from test where name = 'VALUES (a,b,c,d)' ?
It's not an insert query, but matches the Pattern, should use QuotedLexer to check it's an insert query.
This is too complicated for the moment. I have created #16 for tracking

**Pull request merged**